pkg/deploy/api/v1beta3/conversion.go

```diff
@@ -195,6 +197,61 @@ func convert_api_DeploymentCauseImageTrigger_To_v1beta3_DeploymentCauseImageTrig
 	}
 	return nil
 }
+
+func convert_v1beta3_RollingDeploymentStrategyParams_To_api_RollingDeploymentStrategyParams(in *RollingDeploymentStrategyParams, out *newer.RollingDeploymentStrategyParams, s conversion.Scope) error {
+	out.UpdatePeriodSeconds = in.UpdatePeriodSeconds
+	out.IntervalSeconds = in.IntervalSeconds
+	out.TimeoutSeconds = in.TimeoutSeconds
+	out.UpdatePercent = in.UpdatePercent
+
+	if in.UpdatePercent != nil {
```

Comment:
Alright, reading this the `UpdatePercent` is roundtripped through clients.  Unless clients know to set this value to `nil`, they'll never be able to make their `MaxSurge` value take, right?